tests/test_authentication.py

```diff
@@ -118,15 +142,58 @@ def test_invalid_grant_exception(auth_state):
         auth_state.get_authorizer_for_scope("doesn't matter")
 
 
-def test_dependent_token_callout_500_fails_dependent_authorization(auth_state):
-    """On a 5xx response, getting an authorizer fails."""
-    RegisteredResponse(
-        service="auth", path="/v2/oauth2/token", method="POST", status=500
-    ).add()
-    with pytest.raises(globus_sdk.GlobusAPIError):
-        auth_state.get_authorizer_for_scope(
-            "urn:globus:auth:scope:groups.api.globus.org:view_my_groups_and_memberships"
-        )
+@pytest.mark.parametrize(
+    "statuses, succeeds",
+    (
+        pytest.param((500, 200), True, id="500-200-succeeds"),
+        pytest.param((500, 500, 200), False, id="500-500-200-fails"),
+    ),
+)
+def test_dependent_token_callout_500_retry_behavior(
+    caplog, get_auth_state_instance, statuses, succeeds
+):
+    """
+    Test the behavior of 500s and retries using the FastRetryClientFactory defined above.
+
+    On a 5xx response followed by a 200, getting an authorizer succeeds.
+    (Default configuration)
+
+    On two 5xx responses followed by a 200, getting an authorizer fails.
+    (Default configuration)
+    """
+    auth_state = get_auth_state_instance(
+        ["expected-scope"], client_factory=FastRetryClientFactory()
+    )
+    body_for_200s = [
+        {
+            "access_token": "oompa-loompa-doompa-de-access-token",
+            "resource_server": "wonka-chocolates.api.globus.org",
+            "scope": "golden-ticket",
+            "expires_in": 3600,
+            "refresh_token": None,
+            "token_type": "bearer",
+        },
+    ]
+
+    for status in statuses:
+        if status == 200:
+            body = body_for_200s
+        else:
+            body = {}
+        RegisteredResponse(
+            service="auth",
+            path="/v2/oauth2/token",
+            method="POST",
+            status=status,
+            json=body,
+        ).add()
+
+    if succeeds:
+        authorizer = auth_state.get_authorizer_for_scope("golden-ticket")
+        assert isinstance(authorizer, globus_sdk.AccessTokenAuthorizer)
+    else:
+        with pytest.raises(globus_sdk.GlobusAPIError):
```

Comment:
I recommend checking what's in the exception (a message, a value, a shape of data, etc).